src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -450,6 +448,69 @@ void OTARequestor::NotifyUpdateApplied(uint32_t version)
     ConnectToProvider(kNotifyUpdateApplied);
 }
 
+CHIP_ERROR OTARequestor::GetDefaultOtaProviderList(AttributeValueEncoder & encoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    return encoder.EncodeList([&](const auto & enc) -> CHIP_ERROR {
+        mDefaultOtaProviderList.ForEachActiveObject([&](ProviderLocation::Type * pl) {
+            err = enc.Encode(*pl);
+            VerifyOrReturnError(err == CHIP_NO_ERROR, Loop::Break);
+            return Loop::Continue;
+        });
+        return err;
+    });
+}
+
+CHIP_ERROR OTARequestor::ClearDefaultOtaProviderList(FabricIndex fabricIndex)
+{
+    // Remove all entries for the fabric index indicated
+    mDefaultOtaProviderList.ForEachActiveObject([&](ProviderLocation::Type * pl) {
+        if (pl->GetFabricIndex() == fabricIndex)
+        {
+            mDefaultOtaProviderList.ReleaseObject(pl);
+        }
+        return Loop::Continue;
+    });
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OTARequestor::AddDefaultOtaProvider(FabricIndex fabricIndex, ProviderLocation::Type const & providerLocation)
+{
+    if (providerLocation.GetFabricIndex() == kUndefinedFabricIndex)
+    {
+        return CHIP_IM_GLOBAL_STATUS(ConstraintError);
+    }
+
+    // Look for an entry with the same fabric index indicated
+    bool fabricIndexExists = false;
+    mDefaultOtaProviderList.ForEachActiveObject([&](auto * pl) {
+        if (pl->GetFabricIndex() == fabricIndex)
+        {
+            fabricIndexExists = true;
+            return Loop::Break;
```

Comment:
Fwiw, could just use the return value of `ForEachActiveObject` to see whether the loop was terminated by Loop::Break or completed, instead of closing over a boolean.  Either way, though.